bin/filecheck.py

```diff
@@ -654,8 +682,8 @@ def process_file(self, file):
         if file.should_copy:
             file.safe_copy()
             file.set_property('copied', True)
-        file.write_log()
-        if file.is_recursive:
+            file.write_log()
+        if file.is_archive:
             self.process_archive(file)
```

Comment:
Nope, this is one of the ugly things about processing archives. I really need to clean up this code. The write_log happens in process_archive, because the file needs to be unpacked first. It might be worth it to switch to a workflow where we identify files that are archives before doing anything else, and then have a completely different workflow for them, rather than calling file.check() on every file and then handling archives separately later.